app/api/items/get_item.go

```diff
@@ -11,106 +11,239 @@ import (
 }
 
 type itemRootNodeNotChapterFields struct {
-	URL          *string `json:"url"`
-	UsesAPI      bool    `json:"uses_api"`
-	HintsAllowed bool    `json:"hints_allowed"`
+	// Nullable; only if not a chapter
+	URL *string `json:"url"`
+	// only if not a chapter
+	UsesAPI bool `json:"uses_api"`
+	// only if not a chapter
+	HintsAllowed bool `json:"hints_allowed"`
 }
 
 type itemChildNode struct {
 	*itemCommonFields
+
+	// required: true
+	String itemString `json:"string"`
+
 	// items_items (child nodes only)
-	Order            int32  `json:"order"`
-	Category         string `json:"category"`
-	AlwaysVisible    bool   `json:"always_visible"`
-	AccessRestricted bool   `json:"access_restricted"`
+
+	// `items_items.iOrder`
+	// required: true
+	Order int32 `json:"order"`
+	// `items_items.sCategory`
+	// required: true
+	Category string `json:"category"`
+	// `items_items.bAlwaysVisible`
+	// enum: Undefined,Discovery,Application,Validation,Challenge
+	// required: true
+	AlwaysVisible bool `json:"always_visible"`
+	// `items_items.bAccessRestricted`
+	// required: true
+	AccessRestricted bool `json:"access_restricted"`
+
+	// from `users_items`
+	// required: true
+	User itemUser `json:"user"`
 }
 
+// swagger:model itemResponse
 type itemResponse struct {
 	*itemCommonFields
 
 	// root node only
-	TitleBarVisible bool   `json:"title_bar_visible"`
-	ReadOnly        bool   `json:"read_only"`
-	FullScreen      string `json:"full_screen"`
-	ShowSource      bool   `json:"show_source"`
-	ValidationMin   *int32 `json:"validation_min"`
-	ShowUserInfos   bool   `json:"show_user_infos"`
-	ContestPhase    string `json:"contest_phase"`
+
+	// required: true
+	TitleBarVisible bool `json:"title_bar_visible"`
+	// required: true
+	ReadOnly bool `json:"read_only"`
+	// required: true
+	// enum: forceYes,,forceNo,default
+	FullScreen string `json:"full_screen"`
+	// required: true
+	ShowSource bool `json:"show_source"`
+	// Nullable
+	// required: true
+	ValidationMin *int32 `json:"validation_min"`
+	// required: true
+	ShowUserInfos bool `json:"show_user_infos"`
+	// required: true
+	// enum: Running,Analysis,Closed
+	ContestPhase string `json:"contest_phase"`
+
+	// required: true
+	User itemUserRoot `json:"user"`
+
+	// required: true
+	String itemStringRoot `json:"string"`
 
 	*itemRootNodeNotChapterFields
+
+	// required: true
 	Children []itemChildNode `json:"children"`
 }
 
+// swagger:operation GET /items/{item_id} items itemView
+// ---
+// summary: Get an item
+// description: Returns data related to the specified item, its children,
+//              and the current user's interactions with them
+//              (from tables `items`, `items_items`, `items_string`, and `users_items`).
+//
+//
+//              * If the specified item is not visible by the current user, the 'not found' response is returned.
+//
+//              * If the current user has only grayed access on the specified item, the 'forbidden' error is returned.
```

Comment:
why is there field with "only if not greyed" then? 🤔

Reply:
The root item cannot be grayed, but children still can